From 259544c4f4b47418885075c9531ce593c600401c Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <leio@gentoo.org>
Date: Fri, 1 Mar 2019 00:02:47 +0200
Subject: [PATCH] build: Fix build without gles headers

This is an ugly quick fix to hopefully fix non-wayland builds against
mesa[-gles2].
It assumes gnome-shell package USE=wayland requires USE=gles2, which it
sort of does already at runtime already for hybrid graphics support, but
we ensure with REQUIRED_USE.
In meson upstream port for 3.32 this will be more properly conditional
on the gles2 option instead.
---
 src/Makefile.am | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index bcb3505c7..a68661b21 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -118,9 +118,6 @@ libmutter_@LIBMUTTER_API_VERSION@_la_SOURCES =	\
 	backends/meta-egl.c			\
 	backends/meta-egl.h			\
 	backends/meta-egl-ext.h			\
-	backends/meta-gles3.c			\
-	backends/meta-gles3.h			\
-	backends/meta-gles3-table.h		\
 	backends/meta-gpu.c			\
 	backends/meta-gpu.h			\
 	backends/meta-display-config-shared.h	\
@@ -472,6 +469,9 @@ endif
 
 if HAVE_NATIVE_BACKEND
 libmutter_@LIBMUTTER_API_VERSION@_la_SOURCES +=		\
+	backends/meta-gles3.c			\
+	backends/meta-gles3.h			\
+	backends/meta-gles3-table.h		\
 	backends/native/meta-backend-native.c		\
 	backends/native/meta-backend-native.h		\
 	backends/native/meta-backend-native-private.h	\
-- 
2.17.0

